### DISCUSSION TITLE
Why do you need an API key to use this app?

### DISCUSSION THREAD

**Discussion by JRDK92** [L4TTiCe/BotForge#12] [Q&A] [answered]
On Android it asked for an API key. I have not seen any other apps that require an API key to use them. Why this one specifically?

**Comment by dheshanm** [comment, marked as answer]
Hi, This App uses Open AI's API to provide Chat Completions, which is paid, with Pricing information available at [Open AI Pricing] (https://openai.com/pricing/)

As far as I am aware, other Apps may use be using some of the following ways:
- Use other free API services (that are not from Open AI directly) or their own self-hosted LLMs
- Use other means to scrape from official web client
- Use the developer's own API key, in which case the developer may be paying for the API and relying on Ads as a source of income to pay for it
  - I also think there is a rate limit on the API, which makes it even more tricky

Since this App isn't Ad supported, and is an Open AI Client for Android, it relies on the users to use their own API key to make API calls.
